### ISSUE TITLE
use repo2docker

### ISSUE THREAD

**Issue by rokroskar** [SwissDataScienceCenter/renku#406]
If we want to support multiple different environments, we start to need a templated image build. repo2docker does this well, but the images seem a bit bloated and they clone the repo inside the image which is not what we want to do. I think we should consider using repo2docker if we can:

- [ ] avoid cloning the repo (this should be easy and a PR to upstream
- [ ] use it as a more general image build, not notebook-specific (the defaults always launch a notebook)

**Comment by leafty** [comment]
Mmmh I would offer multiple choices of base dockerfile, like lab/scala, lab/r, rstudio, lab/julia, etc.

I suspect we would have to modify repo2docker significantly to have it work for ud

**Comment by rokroskar** [comment]
yeah it's possible. We could build up our own templating of course. I just figured starting out with repo2docker would be quick and we can experiment with how we want the user experience to look from the UI perspective while we refactor the image building.
